custom/icds_reports/static/js/directives/download/download.directive.js

```diff
@@ -345,9 +347,35 @@ function DownloadController($rootScope, $location, locationHierarchy, locationsS
     vm.onSelectYear = function (year) {
         var date = new Date();
         var latest = date;
+        vm.years = vm.yearsCopy;
+        vm.months = vm.monthsCopy;
+
         if (vm.isIncentiveReportSelected()) {
-            var offset = date.getDate() < 15 ? 2 : 1;
-            latest.setMonth(date.getMonth() - offset);
+            vm.years = _.filter(vm.yearsCopy, function (y) {
+                return y.id >= 2018;
+            });
+
+            if (vm.selectedYear === latest.getFullYear()) {
+                var offset = date.getDate() < 15 ? 1 : 0;
+
+                vm.months = _.filter(vm.monthsCopy, function (month) {
+                    return month.id <= (latest.getMonth() + 1) - offset;
+                });
+
+                if (vm.months.length > 9) {
+                    vm.months = vm.months.slice(9);
+                    vm.selectedMonth = 10;
+                } else {
+                    vm.months = [];
+                    vm.selectedMonth = undefined;
+                }
+            } else {
+                vm.months = vm.months.slice(-3);
+                if (vm.selectedMonth === undefined || vm.selectedMonth < 10) {
```

Comment:
This looks like if it we are in 2020 and open the report with date 2019 08. it will not let open. it will make 2019 10. @dhiraj-beehyv  Can you check

Reply:
@Rohit25negi ,
You are right, I misunderstood the requirement and made it available for month October to December. I'll update the pr with the proper fix.

Reply:
@Rohit25negi ,

Updated the pr with the fixes.